From e8fb227b164fcb8dcb7b5a229b61d744a60121e3 Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Mon, 12 Dec 2011 15:10:12 +0530
Subject: [PATCH 100/609] arm:omap:am33xx: Register SGX device

Original commit: bfe8a812aa4b40e63d7967bc4466fb718a725689

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

If SGX is enabled in hardware (information obtained from
device feature register) register SGX platform device.

Signed-off-by: Afzal Mohammed <afzal@ti.com>
---
 arch/arm/mach-omap2/devices.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/devices.c b/arch/arm/mach-omap2/devices.c
index 1be0478..e34ac2d 100644
--- a/arch/arm/mach-omap2/devices.c
+++ b/arch/arm/mach-omap2/devices.c
@@ -47,6 +47,7 @@
 #include <plat/omap_device.h>
 #include <plat/omap4-keypad.h>
 #include <plat/config_pwm.h>
+#include <plat/cpu.h>
 
 /* LCD controller similar DA8xx */
 #include <video/da8xx-fb.h>
@@ -1373,6 +1374,11 @@ void omap_init_pwmss(void)
 	am335x_register_ecap();
 }
 
+static struct platform_device am335x_sgx = {
+	.name	= "sgx",
+	.id	= -1,
+};
+
 #endif
 
 /*-------------------------------------------------------------------------*/
@@ -1401,6 +1407,8 @@ static int __init omap2_init_devices(void)
 #if defined (CONFIG_SOC_OMAPAM33XX)
 	am335x_register_pruss_uio(&am335x_pruss_uio_pdata);
 	omap_init_pwmss();
+	if (omap3_has_sgx())
+		platform_device_register(&am335x_sgx);
 #endif
 	return 0;
 }
-- 
1.7.5.4

